components/browser/session/src/main/java/mozilla/components/browser/session/LegacySessionManager.kt

```diff
@@ -516,6 +516,14 @@ class LegacySessionManager(
         values.find { session -> session.id == id }
     }
 
+    /**
+     * Finds and returns the first session with the given url. Returns null if no matching session could be
+     * found.
+     */
+    fun findSessionByUrl(url: String): Session? = synchronized(values) {
+        values.firstOrNull { session -> session.url == url }
+    }
```

Comment:
Would it be possible to implement this as an extension function in `browser-state` on `BrowserState` instead? If possible I would like to avoid adding anything new to `SessionManager` anymore since we are trying to get rid of it.

Even here I think you would want to maybe limit it to `sessions` instead of `values` to not search in custom tabs and web apps?

Reply:
One question: Presumably I get an  `existingTab`:`TabSessionState`. How can I use that to navigate to the opened tab? 
Without using something like :
`sessionManager.findSessionById(existingTab.id)?.let {
                sessionManager.select(it)
}`

Reply:
Yeah, we'd still need this workaround "translation" code. That will be something we can get rid of once we start migrating Fenix code from `SessionManager` to `BrowserStore`. But at least we do not add new functionality that will end up getting called in various places makign things harder to migrate. :)

Reply:
I agree, I just wanted to see if there is something already migrated so I could avoid using `sessionManager` in yet another place in Fenix. Thank you for the details!